File: ruby/leet_code/1700.rb
# @param {Integer[]} students
# @param {Integer[]} sandwiches
# @return {Integer}
def count_students(students, sandwiches)
  students_count = students.group_by(&:itself).transform_values(&:count)
  sandwiches_count = sandwiches.group_by(&:itself).transform_values(&:count)
  students_count.each do |key, value|
    if sandwiches_count[key].nil?
      return value
    end
    if value > sandwiches_count[key]
      return sandwiches_count[key]
    end
  end
  return 0
end

File: typescript/leet_code/1700.js
function countStudents(students, sandwiches) {
    var totalStudents = students.length;
    var sandwichPointer = 0;
    var unsuccessfulAttempts = 0;
    while (students.length > 0 && unsuccessfulAttempts < totalStudents) {
        var student = students.shift();
        if (student === sandwiches[sandwichPointer]) {
            sandwichPointer++;
            unsuccessfulAttempts = 0;
        }
        else {
            students.push(student);
            unsuccessfulAttempts++;
        }
    }
    return students.length;
}
